consensus/beacon/consensus.go

```diff
@@ -478,8 +483,18 @@ func (beacon *Beacon) SealHash(header *types.Header) common.Hash {
 // the difficulty that a new block should have when created at time
 // given the parent block's time and difficulty.
 func (beacon *Beacon) CalcDifficulty(chain consensus.ChainHeaderReader, time uint64, parent *types.Header) *big.Int {
-	// Transition isn't triggered yet, use the legacy rules for calculation
-	if reached, _ := IsTTDReached(chain, parent.Hash(), parent.Number.Uint64()); !reached {
+	// The beacon engine requires access to total difficulties to be able to
+	// seal pre-merge and post-merge blocks. With the transition to removing
+	// old blocks, TDs become unaccessible, thus making TTD based pre-/post-
+	// merge decisions impossible.
+	//
+	// We do not need to seal non-merge blocks anymore live, but we do need
+	// to be able to generate test chains, thus we're reverting to a testing-
+	// settable field to direct that.
+	cfg := chain.Config()
+	if beacon.ttdblock != nil && *beacon.ttdblock > parent.Number.Uint64() ||
+		// OP-Stack: transitioned networks must use legacy consensus pre-Bedrock
+		(cfg.IsOptimism() && !cfg.IsBedrock(big.NewInt(int64(time)))) {
```

Comment:
added this optimism config check as well @protolambda

Reply:
This is where I introduced a bug. Hit the footgun that Bedrock is activated by block number, not time. Fixed now, and tests are passing.